server/src/main/java/io/crate/expression/scalar/string/StringSplitPartFunction.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+/**
+ * String split_part(String text, String delimiter, Integer field)
+ * <p>
+ * Splits a text on delimiter and returns the and returns the part at index field (1 for the first part)
+ */
+public final class StringSplitPartFunction extends Scalar<String, Object> {
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                "split_part",
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature()
+            ),
+            StringSplitPartFunction::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    public StringSplitPartFunction(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+        assert args.length == 3 : "split_part takes exactly three arguments";
+        var text = (String) args[0].value();
+        var delimiter = (String) args[1].value();
+        var fieldB = (Integer) args[2].value();
```

Comment:
minor: I think naming this var `occurrence` or `positional` makes it more clear.

Reply:
It is named `field` in the PostgreSQL documentation. I changed it to `index` for the Crate documentation but used `field` in the implementation. I just pushed a change to use `index` everywhere. If you prefer `occurrence` or `positional` I can also change it to that.

Reply:
👍 `index` sounds also fine.